packages/ai-jsx/src/core/completion.tsx

```diff
@@ -14,6 +16,18 @@ export type ModelPropsWithChildren = ModelProps & {
 
 export type ModelComponent<T extends ModelPropsWithChildren> = Component<T>;
 
+export type FunctionDefinition = {
```

Comment:
One thing I don't like about this PR is that I have function definitions as props to ChatCompletion as opposed to being children.

I chose to set them as properties because it was hard to parse a nested jsx structure in the children into these types. However, I think it would look much better if they were children because 1) It looks like a more elegant JSX syntax, and 2) A subtree can generate both "SystemMessage" and "FunctionDefinition" for the consumption of this parent.

Reply:
Can you show me a snippet of what the alternative would look like?

Reply:
Something along these lines:
```
<ChatCompletion>
  <Function name="foo">
    This function does the blah operation
    <Parameter name="param1" type="string">
      This parameter controls the blah.
    </Parameter>
    <Parameter name="param2" type="integer">
      Another argument of the function.
    </Parameter>
  </Function>
</ChatCompletion>
```